pkgs/by-name/fi/firebase-tools/001-override-nan.patch

```diff
@@ -0,0 +1,54 @@
+diff --git a/npm-shrinkwrap.json b/npm-shrinkwrap.json
+index da429c128..9c69ec8cb 100644
+--- a/npm-shrinkwrap.json
++++ b/npm-shrinkwrap.json
+@@ -14637,9 +14637,10 @@
+       }
+     },
+     "node_modules/nan": {
+-      "version": "2.17.0",
+-      "resolved": "https://registry.npmjs.org/nan/-/nan-2.17.0.tgz",
+-      "integrity": "sha512-2ZTgtl0nJsO0KQCjEpxcIr5D+Yv90plTitZt9JBfQvVJDS5seMl3FOvsh3+9CoYWXf/1l5OaZzzF6nDm4cagaQ==",
++      "version": "2.22.0",
++      "resolved": "https://registry.npmjs.org/nan/-/nan-2.22.0.tgz",
++      "integrity": "sha512-nbajikzWTMwsW+eSsNm3QwlOs7het9gGJU5dDZzRTQGk03vyBOauxgI4VakDzE0PtsGTmXPsXTbbjVhRwR5mpw==",
++      "license": "MIT",
+       "optional": true
+     },
+     "node_modules/nanoid": {
+@@ -31852,9 +31853,9 @@
+       }
+     },
+     "nan": {
+-      "version": "2.17.0",
+-      "resolved": "https://registry.npmjs.org/nan/-/nan-2.17.0.tgz",
+-      "integrity": "sha512-2ZTgtl0nJsO0KQCjEpxcIr5D+Yv90plTitZt9JBfQvVJDS5seMl3FOvsh3+9CoYWXf/1l5OaZzzF6nDm4cagaQ==",
++      "version": "2.22.0",
++      "resolved": "https://registry.npmjs.org/nan/-/nan-2.22.0.tgz",
++      "integrity": "sha512-nbajikzWTMwsW+eSsNm3QwlOs7het9gGJU5dDZzRTQGk03vyBOauxgI4VakDzE0PtsGTmXPsXTbbjVhRwR5mpw==",
+       "optional": true
+     },
+     "nanoid": {
+@@ -33672,7 +33673,7 @@
+       "optional": true,
+       "requires": {
+         "install-artifact-from-github": "^1.3.1",
+-        "nan": "^2.17.0",
++        "nan": "^2.22.0",
+         "node-gyp": "^9.3.0"
+       }
+     },
+diff --git a/package.json b/package.json
+index a4284d841..90a0524f2 100644
+--- a/package.json
++++ b/package.json
+@@ -98,6 +98,9 @@
+       "src/test/**/*"
+     ]
+   },
++  "overrides" : {
++    "nan": "^2.22.0"
++  },
```

Comment:
not sure if we need this, but it can stay.
(^2.17.0 still accepts 2.22.0, so this change is not strictly required)

Reply:
2.17.0 was causing compilation failures with Node 22 (apparently SetAccessor was renamed in the current v8; see https://github.com/nodejs/nan/blob/main/CHANGELOG.md#2220-oct-11-2024)

Reply:
Reformulating, my main complaint is this: why add an override when you could have just only updated nan with `npm update nan`? Should be the same result and have no change in the package.json

Reply:
Maybe I'm misunderstanding something, but using the `override` makes the change to the locked packages explicit. 

Let's say that a later update changes package.json in a way that doesn't affect ajv (and thus nan). Chances are, they'll regenerate the shrinkwrap from scratch and it would revert to the broken nan@2.17.0. 

Defining the override makes it explicit that we need this later version of nan to support Node 22+.

Am I wrong?

Reply:
I agree what you're doing is better, so LGTM

though even if you have a `^2.17.0` dependency, it's going to get resolved as the latest `2.x` version because of the `^`.
But yes, I see now that we should be more explicit.

Reply:
See my note below. TL;dr -- you're right, this could be an update to the shrinkwrap only. Shall I make that change (removing the override)?